src/main/java/com/cryptomorin/xseries/XItemStack.java

```diff
@@ -768,6 +771,18 @@ public static ItemStack edit(@Nonnull ItemStack item,
                 }
             }
         } else {
+            if (supports(20)) {
+                if (meta instanceof ArmorMeta) {
+                    ArmorMeta armorMeta = (ArmorMeta) meta;
+                    if (config.isSet("trim")) {
+                        ConfigurationSection trim = config.getConfigurationSection("trim");
+                        TrimMaterial trimMaterial = Registry.TRIM_MATERIAL.get(NamespacedKey.minecraft(trim.getString("material")));
+                        TrimPattern trimPattern = Registry.TRIM_PATTERN.get(NamespacedKey.minecraft(trim.getString("pattern")));
```

Comment:
Can you make the namespace configurable as well? `namespace:key` format should do. If no namespace is specified, just use the default minecraft. I don't know if this is a thing in resource packs, but it'll probably be useful for the future.

The `NamespacedKey.fromString(trim.getString(...), "minecraft")` seems to do the job.

Reply:
Suggested changes have been made, I also added armor trims to the serialization process.